### ISSUE TITLE
Reproducibility of ShowUI on ScreenSpot

### ISSUE THREAD

**Issue by Amshaker** [showlab/ShowUI#79]
Dear Authors @QinghongLin @BrightHai @yyyang-2019 @h-siyuan @xyzhu1225 ,

Thank you for making your work open-source and accessible to the CV community; it is truly inspiring and helpful!

I have been working on reproducing the results of the Qwen2-VL-2B-Instruct model. I followed all the steps outlined for data downloading and preprocessing and was able to complete them successfully. Below is the command I used for training on 32 GPUs, as recommended in the paper.

```
deepspeed --include localhost:1 --master_port 5678 train.py \
  --wandb_key=$WANDB_KEY \
  --model_id='Qwen/Qwen2-VL-2B-Instruct' \
  --version='Qwen/Qwen2-VL-2B-Instruct' \
  --dataset_dir=$_DATA_DIR \
  --log_base_dir=$_SAVE_DIR \
  --epochs=50 \
  --steps_per_epoch=100 \
  --batch_size=1 \
  --grad_accumulation_steps=2 \
  --model_max_length=8192 \
  --exp_id="debug" \
  --train_ratio="1,1,1,1,1,1"  \
  --train_dataset="showui-desktop, showui-web, amex, guiact, guiact, guiact,"  \
  --train_json="hf_train, hf_train, hf_train, hf_train_smartphone, hf_train_web-multi, hf_train_web-single"   \
  --val_dataset="screenspot"  \
  --val_json="hf_test"    \
  --precision="bf16" \
  --attn_imple="sdpa" \
  --workers=0 \
  --lora_r=32 \
  --lora_alpha=64  \
  --min_visual_tokens=256  \
  --max_visual_tokens=1344  \
  --num_turn=100 \
  --random_sample \
  --record_sample \
  --lr=0.0001 \
  --uniform_prompt  \
  --ds_zero="zero2" \
  --gradient_checkpointing  \
  --lm_skip_ratio=0.5   \
  --lm_skip_layer='[1,28,0]'    \
  --num_history=2    \
  --interleaved_history='tttt'
```
Despite following the recommended setup, my model achieves 70.6% accuracy on ScreenSpot, whereas the paper reports 75.1%. This 4.5% gap is quite significant, and I would greatly appreciate any insights on potential factors that could contribute to this difference.

- Could there be additional hyperparameters or training nuances not explicitly mentioned? I am also attaching the score of the model across the steps here for your reference.

<img width="374" alt="Image" src="https://github.com/user-attachments/assets/c9055d36-7a60-4ae6-a6f9-b849754d6739" />

Your fast response is highly appreciated.

Best Regards,
Abdelrahman.

**Comment by Han1018** [comment]
I noticed you're using the following training datasets:

```
showui-desktop, showui-web, amex, guiact, guiact, guiact
```
Just wondering — have you ever tried using only:

```
showui-desktop, showui-web, amex
```
for training?
I’m interested in understanding how this smaller combination might affect performance, especially on the ScreenSpot benchmark.

**Comment by Amshaker** [comment]
The performance is decreased slightly.

**Comment by QinghongLin** [comment]
@Han1018 @Amshaker 
Hi In our paper, we provide the score of ShowUI-G, which is based on `showui-desktop, showui-web, amex`. We found score do not have a large change.

**Comment by hhxxttxsh** [comment]
How did you set up the multi-GPU training?
